test/SILOptimizer/return.swift

```diff
@@ -94,6 +94,12 @@ func whileLoop(flag: Bool) -> Int {
   }
 } //expected-error {{missing return in a function expected to return 'Int'}}
 
+struct S {}
+extension S:ExpressibleByStringLiteral {
+  init!(stringLiteral:String) {
+  } // expected-error {{missing return in a function expected to return 'S!'}}
```

Comment:
The real error is that we're missing a self.init delegation or self assignment, but I guess before we weren't diagnosing anything at all here?

Reply:
We were crashing with `missing case` before.

Reply:
I see. LGTM!